### ISSUE TITLE
[Feature] Total lines and position in buffer?

### ISSUE THREAD

**Issue by ReneFroger** [Malabarba/smart-mode-line#152]
I was quite charmed of [Spaceline](https://github.com/TheBB/spaceline). But don't worry, I love the `smart-mode-line` much more, especially the prefixes for directories. 

But I'm missing features in Smart Mode Line, after trying the Spaceline. 

I know you can implement the total lines of the buffer in SML, but in Spaceline it's nicely rounded up: 

![test1](https://cloud.githubusercontent.com/assets/5211845/10200256/8bb0af34-67a5-11e5-9e8d-a05559618877.png)

And much more important missing feature, is see in which positon of the buffer you currently are.  For example, in SML when you're near the end or the top, you see `Bot` or `Top`, which I found it confusing. So I was wondering if this could be disabled in the customization menu? 
And I tried to implement he buffer position indicator implemented in Smart Mode Line, from this: 

![test2](https://cloud.githubusercontent.com/assets/5211845/10200328/ed904052-67a5-11e5-82e7-d17c2fbaf007.png)

but nothing appears on my mode line after implementing the code. My Elisp profiency is somewhat limited. So I was wondering if it would be a big deal to having these two features in the Smart Mode Line?

**Comment by Malabarba** [comment]
> For example, you see directly in which positon of the buffer you currently are. For example, in SML when you're near the end or the top, you see Bot or Top, which I found it confusing. So I was wondering if this could be disabled in the customization menu? 

No, there's no configuration option to change those Top/Bot into percentages. That's because SML doesn't implement this functionality, it just hooks onto Emacs functionality and Emacs doesn't support that.

> And I tried to implement he buffer position indicator implemented in Smart Mode Line, from this: ... but nothing appears on my mode line after implementing the code. My Elisp profiency is somewhat limited. So I was wondering if it would be a big deal to having these buffer indicator in the Smart Mode Line?

It would certainly be a welcome feature. What code do you have so far?

**Comment by vyp** [comment]
> For example, in SML when you're near the end or the top, you see Bot or Top, which I found it confusing

@ReneFroger Just as an aside, isn't this the same as in vim? Or is there a way to change this in vim?

**Comment by alexxsun** [comment]
Since from the screenshot, I find some examples showing nn% instead of ALL, BOT,TOP, is it available now? If so, how to use it to show the buffer position in the percentage? Thank you very much!

I find it at line 452 in the sml source code

```
(defcustom sml/position-percentage-format "%p"
  "Format used to display position in the buffer.
```

It seems sml can show the percentage now.

Thanks again.

**Comment by sten0** [comment]
I can confirm Emacs 25.1.1 supports position in percentages.  
![screenshot_20181002_234913](https://user-images.githubusercontent.com/11372289/46389138-40420100-c69e-11e8-9aa1-74a2f3466779.png)

Beyond that, I'm not sure how to get the visual "loadbar" widget...  Beyond that these seem like ok alternatives, though I've never been comfortable with minimaps:
https://github.com/zk-phi/sublimity
https://github.com/dengste/minimap
